src/material-experimental/mdc-slide-toggle/slide-toggle.scss

```diff
@@ -31,32 +38,18 @@
     }
   }
 
-  // The thumb-underlay element has `mat-mdc-focus-indicator` which sets
-  // a relative position. This element must have absolute positioning. This
-  // has increased specificity than the style set in MDC to guarantee that
-  // it will be absolutely positioned.
-  .mdc-switch__thumb-underlay {
-    position: absolute;
-  }
-
-  // The MDC switch styles related to the hover state are intertwined with the MDC ripple styles.
-  // We currently don't use the MDC ripple due to size concerns, therefore we need to add some
-  // additional styles to restore the hover state.
-  // We use `::after`, rather than `::before`, because `::before` is used by the
-  // generic strong focus indication styles.
-  .mdc-switch__thumb-underlay::after {
-    border-radius: 50%;
+  .mdc-switch__ripple::after {
     content: '';
     opacity: 0;
   }
 
-  .mdc-switch:hover .mdc-switch__thumb-underlay::after {
+  .mdc-switch:hover .mdc-switch__ripple::after {
```

Comment:
It would be great to use `mdc-switch.$ripple-target` anywhere the `.mdc-switch__ripple` selector is used.

Avoiding `.mdc-*` class selectors will help us avoid breaking changes in the future if the class names need to change. In places where they need to be used (such as ripple), we try to expose a variable for them.

It also looks like the `.mat-mdc-switch` class was removed from the root element. I would add and use that instead of `.mdc-switch` for the same reasons: we strongly discourage directly using `.mdc-*` selectors.

```scss
.mat-mdc-switch #{mdc-switch.$ripple-target}::after {
  opacity: map.get(mdc-ripple.$dark-ink-opacities, hover);
  transition: mdc-switch-transition-enter(opacity, 0, 75ms);
}
```